build/templates/__init__.py.mako

```diff
@@ -22,3 +22,80 @@ from ${module_name}.${c['file_name']} import ${c['python_name']}  # noqa: F401
 from ${module_name}.${c['file_name']} import ${c['ctypes_type']}  # noqa: F401
 % endfor
 
+
+python_package_list = 'Python Installed Packages'
+
+
+def get_diagnostic_information():
+    '''Get diagnostic information about the system state that is suitable for printing or logging
+
+    returns: dict
+    '''
+    import os
+    import pkg_resources
+    import platform
+    import struct
+    import sys
+
+    def is_python_64bit():
+        return (struct.calcsize("P") == 8)
+
+    def is_os_64bit():
+        return platform.machine().endswith('64')
+
+    def is_venv():
+        return 'VIRTUAL_ENV' in os.environ
+
+    info = {}
+    if platform.system() == 'Windows':
+        try:
+            import winreg as winreg
+        except ImportError:
+            import _winreg as winreg
+
+        os_name = 'Windows'
+        try:
+            driver_version_key = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, r"SOFTWARE\National Instruments\${config['driver_name']}\CurrentVersion")
+            driver_version = winreg.QueryValueEx(driver_version_key, "Version")[0]
+        except WindowsError:
+            driver_version = 'Unknown'
+    elif platform.system() == 'Linux':
+        os_name = 'Linux'
+        driver_version = 'Unknown'
+    else:
+        os_name = 'Unknown'
+        driver_version = 'Unknown'
+
+    installed_packages = pkg_resources.working_set
+    installed_packages_list = [{'name': i.key, 'version': i.version, } for i in installed_packages]
+
+    info['OS Name'] = os_name
```

Comment:
Why not have:

info['os']['name']
info['os']['version']
info['os']['bitness']
info['driver']['name']
info['driver']['version']
info['module']['name']
info['module']['version']
info['python']['version']
info['python']['bitness']
info['python']['is_venv'] # make it boolean
...


Anyway, the idea is you have logical grouping for things at the expense of decreased human readability, arguably.

Reply:
Changed to this.

Also required changing the print method